File: demo_comm/src/demo_comm_pkg/src/demo_pub_msg.cpp
#include "ros/ros.h"
#include "demo_comm_pkg/Example.h" //倒入自定义信息头文件
#include "ctime"   

int main(int argc, char **argv)
{
    //如何发布消息
    ros::init(argc, argv, "demo_pub_msg"); //初始化节点
    ros::NodeHandle n; //创建句柄

    ros::Publisher pub = n.advertise<demo_comm_pkg::Example>("example", 1000); //创建发布者，并指定话题名和队列大小
    demo_comm_pkg::Example msg; //创建消息

    msg.age=18;
    msg.name="LuoYang";
    msg.height=180;

    ros::Rate loop_rate(10); //设置发布循环频率
    time_t now; //获取当前时间

    while(ros::ok())
    {
        time_t date= time(&now);

        // 发布消息
        pub.publish(msg);
        ROS_INFO(" name:%s  age:%d   height:%f",msg.name,msg.age,msg.height);
        
        msg.age+=1;
        msg.height+=0.1;

        loop_rate.sleep(); //休眠
        ros::spinOnce();
    }
    return 0;
}

File: demo_comm/src/demo_comm_pkg/scripts/demo_pub_msg.py
import rospy
from demo_comm_pkg.msg import Example


if __name__ == '__main__':
    # Initialize the node and create a handle to it
    rospy.init_node('demo_pub_msg')
    # Create a publisher, publish messages at 10Hz
    pub = rospy.Publisher('Example', Example, queue_size=10)
    # Create a rate object to control the loop
    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        #键盘输入信息
        msg=Example()
        msg.name="Yang"
        msg.age=33
        msg.height=176
        
        pub.publish(msg)
        rate.sleep()